Composed/Core/DataStores/MutableDataStore.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+
+public protocol MutableDataStore: DataStore, RandomAccessCollection, RangeReplaceableCollection, MutableCollection { }
+
+extension ArrayDataStore: MutableDataStore {
+
+    public var startIndex: Int {
+        return elements.startIndex
+    }
+
+    public var endIndex: Int {
+        return elements.endIndex
+    }
+
+    public func index(after i: Int) -> Int {
+        return elements.index(after: i)
+    }
+
+    public func index(before i: Int) -> Int {
+        return elements.index(before: i)
+    }
+
+    public subscript(position: Int) -> Element {
+        get { return elements[position] }
+        set { elements[position] = newValue }
+    }
+
+    public func insert<C>(contentsOf newElements: __owned C, at i: Int) where C : Collection, Element == C.Element {
```

Comment:
I'm guessing this has been auto created by Xcode? I don't think `__owned` is something we're meant to use

Reply:
Oh wow, I hadn't even noticed. Yeah I think I used the Fix Me.

Reply:
#24912f8

Reply:
27f88d2805a12916bffa58c809fbaf104832bc83